tests/performance_calculation/test_performance_calculator.py

```diff
@@ -480,3 +482,63 @@ def test_binary_classification_calculate_without_prediction_column():
         _ = calc.calculate(analysis.merge(analysis_targets, on='id'))
     except Exception as exc:
         pytest.fail(f"an unexpected exception occurred: {exc}")
+
+
+def test_oo_behavior_bc_perf_calc(data):  # noqa: D103
```

Comment:
Name could be a bit more descriptive, as this is a weird thing to test.

Maybe something like test_input_dataframes_are_not_altered_by_calculator